src/main/java/com/arpnetworking/metrics/common/sources/CollectdHttpSource.java

```diff
@@ -0,0 +1,276 @@
+import akka.stream.FanInShape2;
+import akka.stream.FlowShape;
+import akka.stream.Graph;
+import akka.stream.Materializer;
+import akka.stream.Supervision;
+import akka.stream.UniformFanOutShape;
+import akka.stream.javadsl.Broadcast;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.GraphDSL;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.stream.javadsl.Zip;
+import akka.util.ByteString;
+import com.arpnetworking.http.RequestReply;
+import com.arpnetworking.metrics.common.parsers.Parser;
+import com.arpnetworking.metrics.common.parsers.exceptions.ParsingException;
+import com.arpnetworking.metrics.mad.model.DefaultRecord;
+import com.arpnetworking.metrics.mad.model.Record;
+import com.arpnetworking.metrics.mad.parsers.CollectdJsonToRecordParser;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.google.common.collect.Maps;
+import net.sf.oval.ConstraintViolation;
+import net.sf.oval.exception.ConstraintsViolatedException;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Processes HTTP posts from Collectd, Extracts data and emits metrics.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class CollectdHttpSource extends ActorSource {
+    @Override
+    protected Props createProps() {
+        return Actor.props(this);
+    }
+
+    /**
+     * Protected constructor.
+     *
+     * @param builder Instance of <code>Builder</code>.
+     */
+    protected CollectdHttpSource(final Builder builder) {
+        super(builder);
+        _parser = builder._parser;
+    }
+
+    private final Parser<List<DefaultRecord.Builder>> _parser;
+
+    /**
+     * Internal actor to process requests.
+     */
+    /* package private */ static final class Actor extends UntypedActor {
+        /**
+         * Creates a {@link Props} for this actor.
+         *
+         * @param source The {@link CollectdHttpSource} to send notifications through.
+         * @return A new {@link Props}
+         */
+        /* package private */ static Props props(final CollectdHttpSource source) {
+            return Props.create(Actor.class, source);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onReceive(final Object message) throws Exception {
+            if (message instanceof RequestReply) {
+                final RequestReply request = (RequestReply) message;
+                Source.single((HttpRequest) request.getRequest())
+                        .via(_processGraph)
+                        .toMat(_sink, Keep.right())
+                        .run(_materializer)
+                        .whenComplete((done, err) -> {
```

Comment:
Closes over the request.  I can create a(n) (anonymous) class each time, or I can leave the lambda.